### DISCUSSION TITLE
MorphOne relation

### DISCUSSION THREAD

**Discussion by ehs4nm** [filamentphp/filament#1600] [Help]
I want to save a relation between a client to the selected user in my ClientResource. Admin would select the corresponding user defined before. But when using BelongsToSelect in resource i get "Return value must be of type ?Illuminate\Database\Eloquent\Relations\BelongsTo", offcopurce it returns a morphOne. How could i get around this?

I have a User model:
```
public function profileable()
    {
      return $this->morphTo();
    }
```
and Client model:
 ```
   public function user()
    {
        return $this->morphOne(User::class, 'profileable');
    }
```
Thanks in advance.

**Comment by danharrin** [comment]
You could copy the BelongsToSelect into a custom class and change it to MorphOne, tweak it as you need to, and share here / PR? Then it can be in the core as a MorphOneSelect.
